autoload/intero/process/backend_info.vim

```diff
@@ -0,0 +1,49 @@
+"""""""""""
+" Version:
+"
+" This file contains functions for parsing and checking backend information,
+" e.g. the GHCi or Intero version.
+"""""""""""
+
+" A backend info contains the backend identifier, which is 'ghci' or 'intero',
+" and a version. A version is a list of three numbers, for the MAJOR, MINOR, and
+" PATCH versions.
+function! intero#process#backend_info#backend_info(backend, major, minor, patch) abort
+    let l:version = [a:major + 0, a:minor + 0, a:patch + 0]
+    return {'backend': a:backend, 'version': l:version}
+endfunction
+
+" Tries to parse the backend info from a list of lines. Returns a dict with
+" backend info if successful, otherwise an empty dict.
```

Comment:
Would be preferable to display an error message, instead of failing silently. Bugs caused by intero-neovim failing to parse something correctly have been among the most annoying to track down.